cmd/awscollector/Dockerfile

```diff
@@ -1,19 +1,59 @@
+# The binary source type (options: build, copy)
+ARG BIN_SRC_TYPE=build
+
 ################################
-#	Building Stage         #	
-#			       #	
+#	Certificate Stage      #
+#			       #
 ################################
-FROM alpine:latest as build 
+FROM alpine:latest AS certs
 
-# update cert
 RUN apk --update add ca-certificates
 
+################################
+#	Building Stage         #
+#			       #
+################################
+FROM golang:1.17 AS prep-build
+
+ENV GOPROXY=direct
```

Comment:
Not really suppose to have this since it is only required by amazon employees but idc you can keep it

Reply:
I think it's fine to keep it.

Reply:
I think this is a significant contributor to the prolonged runtime of the build stages.  Is it possible to extract this to a runtime argument as well so that it can be used only when necessary?

Reply:
Yeah can do. I guess we'd have to use it, but others wouldn't.

Reply:
Added it as an optional build arg with a comment above the declaration. We'll have to include `--build-arg GOPROXY=direct` to be able to build it locally.